pkgs/development/python-modules/pysc2/default.nix

```diff
@@ -0,0 +1,66 @@
+{ buildPythonPackage
+, lib
+, fetchFromGitHub
+, absl-py
+, enum34
+, future
+, futures
+, makeWrapper
+, mock
+, mpyq
+, numpy
+, portpicker
+, protobuf
+, pygame
+, s2clientprotocol
+, six
+, websocket_client
+, sc2-headless
+}:
+
+buildPythonPackage rec {
+  version = "1.2";
+  name = "PySC2-${version}";
+
+  src = fetchFromGitHub {
+    owner = "deepmind";
+    repo = "pysc2";
+    rev = "39f84b01d662eb58b3d95791f59208c210afd4e7";
+    sha256 = "0dfbc2krd2rys1ji75ng2nl0ki8nhnylxljcp287bfb8qyz2m25p";
+  };
+
+  patches = [ ./fix-setup-for-py3.patch ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  propagatedBuildInputs = [
+    absl-py
+    enum34
+    future
+    mock
+    mpyq
+    numpy
+    portpicker
+    protobuf
+    pygame
+    s2clientprotocol
+    six
+    websocket_client
+    sc2-headless
+  ];
+
+  postInstall = ''
+    for f in "$out"/bin/* ; do
+      wrapProgram $f \
+        --set SC2PATH "${sc2-headless}"
+    done
+  '';
```

Comment:
It won't be able to find `sc2` when you use this package as a library and you try to import this package. If this should be supported (and considering you wrote it can be used as a library I think that's the case), then instead of wrapping the executables the path to `sc2` should be hardcoded, e.g. here https://github.com/deepmind/pysc2/blob/c8ad5e8fdc0fff5549c575abea9acbf11d903fe0/pysc2/run_configs/platforms.py#L122

Reply:
Thanks for that. When I looked at the library aspect of the package it did not occur to me that someone could want to launch the environment in their own code instead of invoking one of the bin scripts with their agent as an argument. I've updated it to apply a patch that enables us to substitute the correct path and also cleans up their LD_LIBRARY_PATH manipulations since they aren't necessary in nix.